### PR TITLE
Handle pushing to starterpack screen when unauthed

### PR DESCRIPTION
**Pull request by haileyok**

### CODE REVIEW

File: src/screens/StarterPack/StarterPackScreen.tsx [hunk 1]
```diff
@@ -330,16 +298,41 @@ function Header({
...
+    }
+    const onBeforeRemove = () => {
+      setActiveStarterPack(undefined)
+    }
+
+    navigation.addListener('focus', onFocus)
+    navigation.addListener('beforeRemove', onBeforeRemove)
+
+    return () => {
+      console.log('remove listeners')
```
Comment (gaearon) [inline src/screens/StarterPack/StarterPackScreen.tsx:330]: oops

File: src/screens/StarterPack/StarterPackScreen.tsx [hunk 2]
```diff
@@ -330,16 +298,41 @@ function Header({
...
+
+  React.useEffect(() => {
+    const onFocus = () => {
+      if (hasSession) return
+      setActiveStarterPack({
+        uri: starterPack.uri,
+      })
+    }
+    const onBeforeRemove = () => {
+      setActiveStarterPack(undefined)
```
Comment (gaearon) [inline src/screens/StarterPack/StarterPackScreen.tsx:323]: needs `hasSession` check or no?

File: src/screens/Signup/StepInfo/index.tsx [hunk 3]
```diff
@@ -27,10 +27,18 @@ function sanitizeDate(date: Date): Date {
...
+export function StepInfo({
+  isLoadingStarterPack,
+}: {
+  isLoadingStarterPack: boolean
+}) {
   const {_} = useLingui()
   const {state, dispatch} = useSignupContext()
 
+  useEffect(() => {
+    LayoutAnimation.configureNext(LayoutAnimation.Presets.easeInEaseOut)
```
Comment (gaearon) [inline src/screens/Signup/StepInfo/index.tsx:39]: This is kind of a race condition, effects have no well-defined lifetime in RN right now so this call may align with some different paint. Can we move this closer to the setstate we actually want to animate?

**Review by gaearon — APPROVED** [review]

File: src/screens/Signup/StepInfo/index.tsx (hunk 3 above)
Comment (haileyok) [reply src/screens/Signup/StepInfo/index.tsx:39]: going to leave for now and see per conversation.

**Pull request merged**